src/lib.rs

```diff
@@ -175,6 +181,68 @@ impl<'a> FMBuilder<'a> {
     }
 }
 
+/// An error indicating a failed match.
+/// The pattern and text are copied in so that the error isn't tied to their lifetimes.
+pub struct FMatchError {
+    ptn: String,
+    text: String,
+    fail_line: usize,
+}
+
+impl FMatchError {
+    pub fn failure_line(&self) -> usize {
+        self.fail_line
+    }
+}
+
+impl fmt::Display for FMatchError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // Figure out how many characters are required for the line numbers margin. Rust doesn't
+        // provide the numeric conversions we need, so we are using the `conv` crate here.
+        let max_lines = f64::value_from(usize::max(self.ptn.len(), self.text.len())).unwrap();
```

Comment:
Is this not `f64::try_from` in disguise?

Reply:
It seems you can't convert from a `usize` to a `f64` using the standard lib.

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=21d27089110d024009b2ad6d33ba8559)

I was surprised too.

Reply:
Ouch!